src/Orleans/Providers/ILeaseProvider.cs

```diff
@@ -81,6 +107,23 @@ public class LeaseRequest
         /// Duration of the lease
         /// </summary>
         public TimeSpan Duration { get; set; }
```

Comment:
This is the LeaseRequest class which should be able to be created and modified by the caller. It is marked as immutable(attribute) so it is immutable through the wire